src/Dialect/ONNX/ONNXShapeHelper.cpp

```diff
@@ -280,6 +280,54 @@ LogicalResult ONNXOpBroadcastedShapeHelper::GetAccessExprs(Value operand,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ONNX DepthToSpace Op Shape Helper
+//===----------------------------------------------------------------------===//
+
+ONNXDepthToSpaceOpShapeHelper::ONNXDepthToSpaceOpShapeHelper(
+    ONNXDepthToSpaceOp *newOp)
+    : ONNXOpShapeHelper<ONNXDepthToSpaceOp>(newOp) {}
+
+ONNXDepthToSpaceOpShapeHelper::ONNXDepthToSpaceOpShapeHelper(
+    ONNXDepthToSpaceOp *newOp, ConversionPatternRewriter &rewriter,
+    ArrayValueIndexCapture::GetDenseVal fGetDenseVal,
+    ArrayValueIndexCapture::LoadVal fLoadVal)
+    : ONNXOpShapeHelper<ONNXDepthToSpaceOp>(
+          newOp, rewriter, fGetDenseVal, fLoadVal) {}
+
+LogicalResult ONNXDepthToSpaceOpShapeHelper::Compute(
+    ONNXDepthToSpaceOpAdaptor operandAdaptor) {
+  // Get info about input data operand and blocksize.
+  Value input = operandAdaptor.input();
+  int64_t blocksize = op->blocksize();
+  assert(input.getType().isa<ShapedType>() && "Input should have a shape");
+  assert(blocksize > 0 && "blocksize should be strictly positive");
+
+  int64_t inputRank = input.getType().cast<ShapedType>().getShape().size();
+  assert(inputRank == 4 && "Unexpected input tensor rank");
+
+  // Compute outputDims.
+  // The input tensor has format [N,C,H,W], where N is the batch axis, C is the
+  // channel or depth, H is the height and W is the width. The output tensor has
+  // shape [N, C * blocksize * blocksize, H/blocksize, W/blocksize].
```

Comment:
This comment does not agree with the code. The code appears correct. Please double check and correct the one that is off